src/main/java/lesson2/problem5/Partition.java

```diff
@@ -0,0 +1,43 @@
+package lesson2.problem5;
+
+import lesson2.problem0.Link;
+import lesson2.problem0.LinkedList;
+
+public class Partition {
+
+    public Link partition(Link head, int pivot) {
+        if (head == null ) {
+            return new Link(0);
+        }
+
+        Link before = new Link(0);
+        Link onOrAfter = new Link(0);
+        Link curr1 = before, curr2 = onOrAfter;
+
+        while (head != null) {
+            if (head.value < pivot) {
+                curr1.next = head;
+                curr1 = head;
+            }
+            else {
+                curr2.next = head;
+                curr2 = head;
+            }
+            head = head.next;
+        }
+        curr2.next = null;
+        curr1.next = onOrAfter.next;
+        return before.next;
```

Comment:
I have not tested, but not sure if this implementation will put the pivot element in the middle. Just count how many you have and append it to the end of the before list.